### PR TITLE
ARROW-2531: [C++] Update clang bits to 6.0

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: cpp/cmake_modules/FindClangTools.cmake [hunk 1]
```diff
@@ -77,6 +77,18 @@ if (CLANG_FORMAT_VERSION)
...
         endif()
+
+        if ("${CLANG_FORMAT_BIN}" STREQUAL "CLANG_FORMAT_BIN-NOTFOUND")
+          # binary was still not found, look into Cellar
+          # TODO: This currently only works for '.0' patch releases as
+          #       find_program does not support regular expressions
+          #       in the paths.
+          find_program(CLANG_FORMAT_BIN
+            NAMES clang-format
+            PATHS /usr/local/Cellar/llvm/${CLANG_FORMAT_VERSION}.0/bin
```
Comment (pitrou) [inline cpp/cmake_modules/FindClangTools.cmake]: What is "Cellar"? Is there a reason we want to hardcode this?

Comment (xhochy) [reply cpp/cmake_modules/FindClangTools.cmake]: This is the location where homebrew stores its unpacked bottles. This is a fixed term. The only location that we might want to make configurable is the use of `/usr/local/` here which is default for homebrew but not always the case.

Comment (pitrou) [reply cpp/cmake_modules/FindClangTools.cmake]: I see, thanks.

File: cpp/README.md [hunk 2]
```diff
@@ -320,7 +320,7 @@ You can also fix any formatting errors automatically:
 
     make format
 
-These commands require `clang-format-5.0` (and not any other version).
+These commands require `clang-format-6.0` (and not any other version).
```
Comment (andrioni) [inline cpp/README.md]: Could you also update the `brew install llvm@5` command to `llvm@6` a couple lines below? Just tested it on macOS and it works fine.

Comment (xhochy) [reply cpp/README.md]: Done

**Comment by xhochy** [comment]
Green appveyor: https://ci.appveyor.com/project/xhochy/arrow/build/1.0.662

**Comment by robertnishihara** [comment]
A bit off topic, but one thing that would be really helpful is to show the linting errors in Travis when clang-format fails.

When I want to do linting for Arrow, I often have to build clang-format from source and `make format` never works out of the box for me for whatever reason, so being able to just look at the Travis build would be useful.

**Comment by xhochy** [comment]
@robertnishihara This should come with https://github.com/apache/arrow/pull/1918

**Comment by robertnishihara** [comment]
Amazing, that's a big help.

**Review by xhochy — COMMENTED** [review]
+1

**Pull request closed without merging**